codec/types/amino_compat.go

```diff
@@ -13,10 +16,17 @@ type aminoCompat struct {
 	err    error
 }
 
+const errorExplaination = ", this is likely because UnpackInterfacesMessage is not defined for some type which contains a protobuf Any either directly or via one of its members"
+
+func aminoCompatError(errType string, x interface{}) error {
+	debug.PrintStack()
```

Comment:
Should we be calling this?

Reply:
Yes, I think so because if this error is occurring, it's something that needs fixing and may be hard to find the root cause of. This is what I needed to figure out where `UnpackInterfacesMessage` wasn't implemented and I left this here with a clear error message to help the debugger figure out what to do.

Reply:
Or can you think of any better way to handle this?

Reply:
Set a variable and switch on that:

```go
// Debug enables stack trace printing on serialization errors.
var Debug = false

func aminoCompatError(errType string, x interface{}) error {
  if Debug {
    debug.PrintStack()
  }

  // ...
}
```

Reply:
My preference would be to address this in a way that makes these errors as easy as possible to fix (which requires implementing `UnpackInterfacesMessage` usually). It's expected that they will be encountered and I want to make life as easy as possible for the developer who sees them.

I think pulling the stack trace with `errors.Wrap` adds another layer of complexity of needing to do this: https://godoc.org/github.com/pkg/errors#hdr-Retrieving_the_stack_trace_of_an_error_or_wrapper.

So I would prefer to either leave it as I have it or add the `Debug` flag and make it clear in the error message which file to go to do enable the debug stack trace. Unless there is some other middleware for `errors.Wrap` that will make this equally easy.

Do you have a preference here @alexanderbez ?

Reply:
This is a corner-case. I'd opt for the debug option which defaults to false.

Reply:
Okay, I'll do that